src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9898,8 +9804,17 @@ internal BracketedArgumentListSyntax ParseBracketedArgumentList()
 
                 _termState = saveTerm;
 
-                openToken = open;
-                closeToken = this.EatToken(closeKind);
+                if (this.CurrentToken.Kind == SyntaxKind.CloseParenToken ||
+                    this.CurrentToken.Kind == SyntaxKind.CloseBracketToken)
+                {
+                    // convert `]` into `)` or vice versa for error recovery
+                    closeToken = this.ConsumeOneToken(closeKind);
```

Comment:
>this.ConsumeOneToken(closeKind) [](start = 33, length = 31)

Why consume the ']' or ')' if it's not the close kind expected? Is it more likely the user closed with the wrong kind or that the expected kind is missing?